inventory_reporting/reports/inventory_report.py

```diff
@@ -106,59 +106,52 @@ def generate_storable_report(self, workbook, wizard):
                 "filter": [
                     ("stock_move_id.picking_type_id.code", "=", "incoming"),
                     ("stock_move_id.origin_returned_move_id", "=", False),
+                    ("stock_move_id.unbuild_id", "=", False),
                 ],
             },
             {
                 "name": _("Return"),
                 "filter": [
+                    "|",
+                    ("stock_move_id.picking_type_id.code", "=", "internal"),
+                    ("stock_move_id.picking_type_id.code", "=", "outgoing"),
                     ("stock_move_id.origin_returned_move_id", "!=", False),
-                    ("reference", "ilike", "/OUT/"),
                 ],
             },
             {
                 "name": _("Delivery"),
                 "filter": [
+                    "|",
                     ("stock_move_id.picking_type_id.code", "=", "internal"),
+                    ("stock_move_id.picking_type_id.code", "=", "outgoing"),
+                    ("stock_move_id.unbuild_id", "=", False),
+                    ("stock_move_id.origin_returned_move_id", "=", False),
+                    ("stock_move_id.location_dest_id.usage", "=", "production"),
                 ],
             },
             {
                 "name": _("Inventory Adjustment"),
                 "filter": [
-                    "|",
-                    (
-                        "stock_move_id.location_id",
-                        "ilike",
-                        "Virtual Locations/Inventory adjustment",
-                    ),
-                    (
-                        "stock_move_id.location_dest_id",
-                        "ilike",
-                        "Virtual Locations/Inventory adjustment",
-                    ),
+                    ("stock_move_id.location_id.usage", "!=", "production"),
+                    ("stock_move_id.location_dest_id.usage", "!=", "production"),
+                    ("stock_move_id.picking_id", "=", False),
+                    ("stock_move_id.unbuild_id", "=", False),
+                ],
+            },
+            {
+                "name": _("Scrap"),
+                "filter": [
+                    ("stock_move_id.scrapped", "=", True),
                 ],
             },
             {
                 "name": _("Subcontracting"),
                 "filter": [
                     "|",
-                    (
-                        "stock_move_id.location_id",
-                        "ilike",
-                        "Physical Locations/Subcontracting Location",
-                    ),
-                    (
-                        "stock_move_id.location_dest_id",
-                        "ilike",
-                        "Physical Locations/Subcontracting Location",
-                    ),
+                    ("stock_move_id.location_id.usage", "=", "production"),
+                    ("stock_move_id.location_dest_id.usage", "=", "production"),
```

Comment:
To filter records "subcontracting".  Please check the validity of this suggestion.

```suggestion
                    "&",
                    ("stock_move_id.location_id.usage", "=", "production"),
                    ("stock_move_id.location_dest_id.is_subcontracting_location", "=", True),
                    "&",
                    ("stock_move_id.location_dest_id.usage", "=", "production"),
                    ("stock_move_id.location_id.is_subcontracting_location", "=", True),
```